gap/semigroups/semitrans.gi

```diff
@@ -735,6 +735,47 @@ function(S)
   return t;
 end);
 
+InstallMethod(WreathProduct, "for a monoid and a permutation group",
+[IsTransformationSemigroup, IsPermGroup],
+function(S, G)
+  local maps, gensS, next, reps, n, i, g, x, m;
+  if not IsMonoidAsSemigroup(S) then
+    ErrorNoReturn("S should be a monoid");
+  fi;
+ 
+  m := LargestMovedPoint(G);
+  maps := []; # the final generating set for the wreath product
+
+  gensS := ShallowCopy(GeneratorsOfSemigroup(S));
+  n := DegreeOfTransformationCollection(gensS);
+  for i in [1 .. Length(gensS)] do 
+    gensS[i] := OnTuples([1 .. n], gensS[i]);
+  od;
+  
+  # move copies of S as by the action induced by G
+  next := [1 .. m * n];
+  for g in GeneratorsOfGroup(G) do 
+    for i in [1 .. m] do 
+      next{[1 .. n] + (i ^ g - 1) * n} := gensS[1] + (i - 1) * n;
+    od;
+    Add(maps, Transformation(next));
+  od;
+
+  #If there is only 1 generator there is nothing else to do
+  #If there are more than one generator, copy them to each distinct orbit of G
+  if not Size(gensS) = 1 then
+    reps := List(Orbits(G, [1 .. m], OnPoints), Representative);
+    for i in reps do 
+      next := [1 .. n * m];
+      for x in gensS do
+        next{[1 .. n] + (i - 1) * n} := x + (i - 1) * n;
+        Add(maps, Transformation(next));
+      od;
+    od;
+  fi;
+  return Semigroup(maps);
```

Comment:
Even though it is a monoid, using Monoid(maps) adds the identity transformation, which may not be in the wreath product. That is why using Semigroup(maps) is the appropiate thing to do.